src/app/passengers/components/smart/trip-planner/trip-planner.component.ts

```diff
@@ -167,12 +171,61 @@ export class TripPlannerComponent implements OnInit {
     this.store.dispatch(new MapActions.AddLocation(map));
   }
 
+  getTripEstimate(event){
+    var res;
+
+    const headers = new HttpHeaders({
+      'Content-Type': 'application/x-www-form-urlencoded',
+      'Authorization': 'Basic ' + btoa('front-end:front-end')
+    });
+
+    const data = "grant_type=password&scope=webclient&username=passenger&password=password";
+
+    const options = {
+      headers,
+      withCredentials: true
+    }; 
+
+   this.http.post('http://localhost:32775/auth/oauth/token', data, options).subscribe(res => {
+     if(res){
+       localStorage.setItem("accessToken", JSON.parse(JSON.stringify(res)).access_token)
+     }
+   })
+   this.postToCalculstionService()
+}
+// Request to communicate with calculation service via JWT token
+  postToCalculstionService = function(){
+    const token = localStorage.getItem("accessToken");
+    const headers = new HttpHeaders({
+    'Content-Type': 'application/json',
+    'Authorization': 'Bearer '+ token
+    });
+
+    const resOptions = {
+    headers,
+    withCredentials: true
+    };
```

Comment:
Can we extract lines 198-207 into a separate method? We will likely need to reuse this logic.

Reply:
this has been applied